pkg/config/config.go

```diff
@@ -81,43 +75,171 @@ func Cmd() *cobra.Command {
 
 // FlagTpl setups the template flag to the given cobra command.
 func FlagTpl(cmd *cobra.Command) *string {
-	return cmd.Flags().StringP("template", "t", "", "custom YAML template file")
+	return cmd.Flags().StringP("template", "t", defaultDeploymentTemplate, "Deployment files template directory (TODO: list presets)")
+}
+
+// ValidateTpl validates the provided option.
+func ValidateTpl(tplDirName *string) error {
+	if _, statErr := os.Stat(*tplDirName); statErr == nil {
+		return nil
+	}
+
+	var embeddedPath = filepath.Join("templates", *tplDirName)
+	if _, statErr := fs.Stat(deploymentTemplates, embeddedPath); statErr == nil {
+		return nil
+	}
+	return fmt.Errorf("template file %q neither exists on disk nor is a known templeate", *tplDirName)
+}
+
+// ReadTplFiles reads the provided option and returns template files as a walkable FS
+func ReadTplFiles(tplDirName *string, cfg *YmlConfig) (fs.FS, error) {
+	var tplDirFs fs.FS = nil
+	// If template dir is given and exists, read it
+	if fInfo, statErr := os.Stat(*tplDirName); statErr == nil {
+		if fInfo.IsDir() {
+			tplDirFs = os.DirFS(*tplDirName)
+		} else {
+			// Be compatible when given a single file
+			// -> mock an FS containing only that file
+			var err error
+			var content []byte
+			content, err = os.ReadFile(*tplDirName)
+			if err != nil {
+				return nil, fmt.Errorf("reading template file %q: %w", *tplDirName, err)
+			}
+			tplDirFs = fstest.MapFS{fInfo.Name(): {Data: content}}
+		}
```

Comment:
There may be a better way to do this.
From what I've read `fstest.MapFS` is actually intended for mocking files and dir trees for tests.
This is done to keep compatibility with providing a single file to `--template`